src/otx/core/config/hpo.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Config objects for HPO."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Literal
+
+
+@dataclass
+class HpoConfig:
+    """DTO for HPO configuration."""
+
+    search_space: dict[str, dict[str, Any]] | None = None
+    save_path: str | None = None
+    mode: Literal["max", "min"] = "max"
+    num_trials: int | None = None
+    num_workers: int | None = None
+    num_full_iterations: int | float | None = None
+    non_pure_train_ratio: float | None = None
+    full_dataset_size: int | None = None
+    expected_time_ratio: int | float | None = 4
+    maximum_resource: int | float | None = None
+    subset_ratio: float | int | None = None
+    min_subset_size: int | None = None
+    prior_hyper_parameters: dict | list[dict] | None = None
+    acceptable_additional_time_ratio: float | int = 1.0
+    minimum_resource: int | float | None = None
+    reduction_factor: int | None = None
+    asynchronous_sha: bool = True
```

Comment:
Do not insert `None` to typing if there is no special reason. Remove `None` from typing and change `None` default values to their actual default values.

Reply:
done at https://github.com/openvinotoolkit/training_extensions/pull/2912/commits/cd014b8aff92f9109cd44eeffaae365912b7aca9.
I changed all of them to use default value of `HyperBand` argument except search_space. `search_space` is required arguments for `HyperBand` but when user runs HPO w/o `hpo_config`, learning rate and batch size are set as HPO candidates based on current learning rate and batch size. So I leave it as it is.